lib/config/migrations/custom/unpublish-safe-migration.ts

```diff
@@ -0,0 +1,43 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class UnpublishSafeMigration extends AbstractMigration {
+  private static readonly SUPPORTED_VALUES = [
+    ':unpublishSafe',
+    'default:unpublishSafe',
+    'npm:unpublishSafe',
+  ];
+
+  override readonly deprecated = true;
+  override readonly propertyName = 'unpublishSafe';
+
+  override run(value: unknown): void {
+    const extendsValue = this.get('extends');
+    const newExtendsValue = Array.isArray(extendsValue) ? extendsValue : [];
+
+    if (value === true) {
+      if (is.string(extendsValue)) {
+        newExtendsValue.push(extendsValue);
+      }
+
+      if (newExtendsValue.every((item) => !this.isSupportedValue(item))) {
+        newExtendsValue.push('npm:unpublishSafe');
+      }
+
+      this.setHard(
+        'extends',
+        newExtendsValue.map((item) => {
+          if (this.isSupportedValue(item)) {
+            return 'npm:unpublishSafe';
```

Comment:
```suggestion
            return ':unpublishSafe';
```
@rarkins aren't our `npm` presets only redirects to internal presets?

Reply:
No, seems to be vice versa